src/driver/compile.ml

```diff
@@ -107,252 +77,173 @@ let find_partials odoc_dir =
-          | Some impl -> (
-              match impl.mip_src_info with
-              | Some si ->
-                  let odoc_file = Fpath.(output_dir // impl.mip_odoc_file) in
-                  let odocl_file = Fpath.(linked_dir // impl.mip_odocl_file) in
-                  Odoc.compile_impl ~output_dir ~input_file:impl.mip_path
-                    ~includes ~parent_id:impl.mip_parent_id ~source_id:si.src_id;
-                  Atomic.incr Stats.stats.compiled_impls;
-                  Some
-                    {
-                      impl_odoc = odoc_file;
-                      impl_odocl = odocl_file;
-                      src = si.src_path;
-                    }
-              | None -> None)
-          | None -> None
-        in
-
-        Odoc.compile ~output_dir ~input_file:modty.m_intf.mif_path ~includes
-          ~parent_id:modty.m_intf.mif_parent_id;
+        let includes = Fpath.Set.of_list unit.include_dirs in
+        Odoc.compile ~output_dir:unit.output_dir ~input_file:unit.input_file
+          ~includes ~parent_id:unit.parent_id;
         Atomic.incr Stats.stats.compiled_units;
 
-        let odoc_output_dir = Fpath.split_base odoc_file |> fst in
-
-        Ok
-          {
-            m = Module modty;
-            odoc_output_dir;
-            odoc_file;
-            odocl_file;
-            include_dirs = includes;
-            impl;
-            pkg_args;
-            pkg_dir = modty.m_pkg_dir;
-            pkg_name = package_name;
-          }
+        Ok unit
   in
 
-  let rec compile : string -> (compiled, exn) Result.t =
-   fun hash ->
+  let rec compile_mod :
+      Odoc_unit.intf Odoc_unit.unit ->
+      (Odoc_unit.intf Odoc_unit.unit, exn) Result.t =
+   fun unit ->
+    let hash = match unit.kind with `Intf { hash; _ } -> hash in
     match Hashtbl.find_opt tbl hash with
     | Some p -> Promise.await p
     | None ->
         let p, r = Promise.create () in
         Hashtbl.add tbl hash p;
-        let result = compile_one compile hash in
+        let result = compile_one compile_mod hash in
         Promise.resolve r result;
         result
   in
-  let to_build = Util.StringMap.bindings hashes |> List.map fst in
-  let mod_results = Fiber.List.map compile to_build in
-  let zipped_res = List.map2 (fun a b -> (a, b)) to_build mod_results in
-  let zipped =
-    List.filter_map (function a, Ok b -> Some (a, b) | _ -> None) zipped_res
+  let to_build = Util.StringMap.bindings hashes |> List.map snd in
+  let mod_results = Fiber.List.map compile_mod to_build in
+  let compile_mld (unit : Odoc_unit.mld Odoc_unit.unit) =
+    let includes = Fpath.Set.of_list unit.include_dirs in
+    Odoc.compile ~output_dir:unit.output_dir ~input_file:unit.input_file
+      ~includes ~parent_id:unit.parent_id;
+    Atomic.incr Stats.stats.compiled_mlds
   in
-  let mods =
-    List.filter_map (function Ok x -> Some x | Error _ -> None) mod_results
+  let _compiled_mlds = Fiber.List.map compile_mld mld_units in
```

Comment:
That's a `unit list`. Do you mean to use `Fiber.List.iter` ?

Reply:
Yes, thanks!